classes/DirtEventTypes.sc

```diff
@@ -81,24 +80,23 @@ DirtEventTypes {
 
 				if(midiout.notNil) {
 					if(latency == 0.0) {
-						if (~nrpn.notNil) {
-							midiout.control(~chan, 99, ~nrpnMSB);
-							midiout.control(~chan, 98, ~nrpnLSB);
-							midiout.control(~chan, 6,  ~valMSB);
-							midiout.control(~chan, 38, ~valLSB)
+						if (nrpnMSB.notNil) {
+							midiout.control(chan, 99, nrpnMSB);
+							midiout.control(chan, 98, nrpnLSB);
+							midiout.control(chan, 6,  valMSB);
+							midiout.control(chan, 38, valLSB)
 						}
 					    {
 							midiout.performList(midicmd, args)
 						}
 					}
 					{
-						if (~nrpn.notNil) {
+						if (nrpnMSB.notNil) {
 							thisThread.clock.sched(latency, {
-								midiout.control(0,99,0);
-								midiout.control(~chan, 99, ~nrpnMSB);
-								midiout.control(~chan, 98, ~nrpnLSB);
-								midiout.control(~chan, 6,  ~valMSB);
-								midiout.control(~chan, 38, ~valLSB)
+								midiout.control(chan, 99, nrpnMSB);
```

Comment:
you don't need to duplicate this code. Jus define a function further up and call it in both cases. Functions are first-class, so they are just something like:

```
var sendNRPN = {
	midiout.control(chan, 99, nrpnMSB);
	midiout.control(~chan, 98, ~nrpnLSB);
	midiout.control(chan, 98, nrpnLSB);
	midiout.control(~chan, 6,  ~valMSB);
	midiout.control(chan, 6,  valMSB);
	midiout.control(~chan, 38, ~valLSB)
	midiout.control(chan, 38, valLSB)
};

```

Reply:
Thanks, done!